raiden/network/transport/matrix/utils.py

```diff
@@ -387,126 +396,134 @@ def join_broadcast_room(client: GMatrixClient, broadcast_room_alias: str) -> Roo
         )
 
 
-def login_or_register(
-    client: GMatrixClient, signer: Signer, prev_user_id: str = None, prev_access_token: str = None
-) -> User:
-    """Login to a Raiden matrix server with password and displayname proof-of-keys
+def first_login(client: GMatrixClient, signer: Signer, username: str) -> User:
+    """Login for the first time.
```

Comment:
It's not necessarily only the first login.
There are multiple cases where a previous auth token can become invalid:
- The server is configured to automatically invalidate tokens after a while (not the default)
- A server operator may manually wipe or invalidate existing access tokens
- A node may have roamed to a different server (e.g. because the original server was temporarily unavailable) and is now 'returning' to the previously used server.

This is mostly a semantic difference though, the code and logic should be fine as is.